system/db/exp.txt

```diff
@@ -639,36 +639,36 @@
 	{ rank:  4, level: 12, exp: 69847584 },
 	{ rank:  4, level: 13, exp: 77197088 },
 	{ rank:  4, level: 14, exp: 84665312 },
-	{ rank:  4, level: 15, exp: 84665312 },
-	{ rank:  4, level: 16, exp: 102995008 },
+	{ rank:  4, level: 15, exp: 99314432 },
+	{ rank:  4, level: 16, exp: 3680576 },
 	{ rank:  4, level: 17, exp: 25764096 },
 	{ rank:  4, level: 18, exp: 47847680 },
 	{ rank:  4, level: 19, exp: 69985280 },
 	{ rank:  4, level: 20, exp: 497258368 },
-	{ rank:  4, level: 21, exp: 519341824 },
-	{ rank:  4, level: 22, exp: 541479424 },
-	{ rank:  4, level: 23, exp: 563563008 },
-	{ rank:  4, level: 24, exp: 585646592 },
-	{ rank:  4, level: 25, exp: 1012973824 },
-	{ rank:  4, level: 26, exp: 1035057152 },
-	{ rank:  4, level: 27, exp: 1057140736 },
-	{ rank:  4, level: 28, exp: 1079278592 },
-	{ rank:  4, level: 29, exp: 1101361664 },
-	{ rank:  4, level: 30, exp: 1436537856 },
-	{ rank:  4, level: 31, exp: 5326848 },
-	{ rank:  4, level: 32, exp: 37293056 },
-	{ rank:  4, level: 33, exp: 69257216 },
-	{ rank:  4, level: 34, exp: 101221376 },
-	{ rank:  4, level: 35, exp: 719555584 },
-	{ rank:  4, level: 36, exp: 751519744 },
-	{ rank:  4, level: 37, exp: 783484928 },
-	{ rank:  4, level: 38, exp: 815518720 },
-	{ rank:  4, level: 39, exp: 847483904 },
-	{ rank:  4, level: 40, exp: 1465817088 },
-	{ rank:  4, level: 41, exp: 1497781248 },
-	{ rank:  4, level: 42, exp: 1529747456 },
-	{ rank:  4, level: 43, exp: 1561712640 },
-	{ rank:  4, level: 44, exp: 1593675776 },
+	{ rank:  4, level: 21, exp: 519341820 },
```

Comment:
I believe we have a bit of a problem here. The level 15 issue appears correctly fixed, and I found where the bug originated, but these very high values, or rather the differences in our numbers, seem to stem from floating point imprecision. I can't tell which numbers are correct right now, because every floating point converter gives me slightly different ones. Have you checked these against what the client displays for these levels by any chance?

Reply:
I checked the client's behavior when setting certain EXP amounts, and it seems like our current values are correct for levels 21 and above. Or rather, our interpretation of the data matches that of the client. So the changes to this range would have to be reverted for the EXP amounts to be correct.